src/api/db/data/20250131094818_nullify_targets_on_bs_request_actions.rb

```diff
@@ -3,30 +3,12 @@
 class NullifyTargetsOnBsRequestActions < ActiveRecord::Migration[7.0]
   # rubocop:disable Rails/SkipsModelValidations
   def up
-    bs_request_actions = BsRequestAction.where('target_project_id IS NOT NULL AND target_package_id IS NULL')
-    bs_request_actions.in_batches do |batch|
-      batch.find_each do |action|
-        target_project = Project.find_by(name: action.target_project)
-        if target_project.nil?
-          action.update_columns(target_project_id: nil)
-        end
-      end
+    BsRequestAction.joins('LEFT JOIN projects ON bs_request_actions.target_project_id = projects.id WHERE bs_request_actions.target_project_id IS NOT NULL AND projects.id IS NULL').in_batches do |batch|
+      batch.update_all(target_project_id: nil)
```

Comment:
```suggestion
    BsRequestAction.joins('LEFT JOIN projects ON bs_request_actions.target_project_id = projects.id WHERE bs_request_actions.target_project_id IS NOT NULL AND projects.id IS NULL')
                   .in_batches
                   .update_all(target_project_id: nil)
```
If we are not going to perform any other action in each batch, we can get rid of the `do |batch|` clause.

Just following the [reference documentation](https://api.rubyonrails.org/v7.0/classes/ActiveRecord/Batches.html#method-i-in_batches).

Same thing for the other query.